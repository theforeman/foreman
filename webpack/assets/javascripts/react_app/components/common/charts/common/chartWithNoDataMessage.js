import React from 'react';
import MessageBox from '../../MessageBox';

export const chartWithNoDataMessage = (WrappedComp, noDataMsg = 'No data available') => (props) => {
  const columns = props && props.data && props.data.columns ? props.data.columns : [];
  if (columns.length > 0) {
    return (<WrappedComp {...props} />);
  }
  return <MessageBox msg={noDataMsg} icontype="info" />;
};
