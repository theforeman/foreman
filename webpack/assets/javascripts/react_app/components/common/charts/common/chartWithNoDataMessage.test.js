import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import { chartWithNoDataMessage } from './chartWithNoDataMessage';

describe('common chart components', () => {
  const Chart = chartWithNoDataMessage(() => <h1>Chart</h1>);
  it('render a messagebox if no data', () => {
    [undefined, null, {}, { data: {} }, { data: { columns: [] } }]
      .forEach(props => expect(toJson(mount(<Chart {...props} />))).toMatchSnapshot());
  });
  it('render chart if there is data', () => {
    expect(toJson(mount(<Chart
      data={{ columns: [['time', 1, 2], ['data', 0, 20]] }} />))).toMatchSnapshot();
  });
});
