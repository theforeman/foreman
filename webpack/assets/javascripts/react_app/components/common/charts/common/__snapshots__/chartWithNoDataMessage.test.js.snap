// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common chart components render a messagebox if no data 1`] = `
<Component>
  <MessageBox
    icontype="info"
    msg="No data available"
  >
    <div
      className="message-box-root"
    >
      <div
        className="pficon pficon-info message-box-content message-box-icon"
      />
      <div
        className="message-box-content message-box-message"
      >
        No data available
      </div>
    </div>
  </MessageBox>
</Component>
`;

exports[`common chart components render a messagebox if no data 2`] = `
<Component>
  <MessageBox
    icontype="info"
    msg="No data available"
  >
    <div
      className="message-box-root"
    >
      <div
        className="pficon pficon-info message-box-content message-box-icon"
      />
      <div
        className="message-box-content message-box-message"
      >
        No data available
      </div>
    </div>
  </MessageBox>
</Component>
`;

exports[`common chart components render a messagebox if no data 3`] = `
<Component>
  <MessageBox
    icontype="info"
    msg="No data available"
  >
    <div
      className="message-box-root"
    >
      <div
        className="pficon pficon-info message-box-content message-box-icon"
      />
      <div
        className="message-box-content message-box-message"
      >
        No data available
      </div>
    </div>
  </MessageBox>
</Component>
`;

exports[`common chart components render a messagebox if no data 4`] = `
<Component
  data={Object {}}
>
  <MessageBox
    icontype="info"
    msg="No data available"
  >
    <div
      className="message-box-root"
    >
      <div
        className="pficon pficon-info message-box-content message-box-icon"
      />
      <div
        className="message-box-content message-box-message"
      >
        No data available
      </div>
    </div>
  </MessageBox>
</Component>
`;

exports[`common chart components render a messagebox if no data 5`] = `
<Component
  data={
    Object {
      "columns": Array [],
    }
  }
>
  <MessageBox
    icontype="info"
    msg="No data available"
  >
    <div
      className="message-box-root"
    >
      <div
        className="pficon pficon-info message-box-content message-box-icon"
      />
      <div
        className="message-box-content message-box-message"
      >
        No data available
      </div>
    </div>
  </MessageBox>
</Component>
`;

exports[`common chart components render chart if there is data 1`] = `
<Component
  data={
    Object {
      "columns": Array [
        Array [
          "time",
          1,
          2,
        ],
        Array [
          "data",
          0,
          20,
        ],
      ],
    }
  }
>
  <Component
    data={
      Object {
        "columns": Array [
          Array [
            "time",
            1,
            2,
          ],
          Array [
            "data",
            0,
            20,
          ],
        ],
      }
    }
  >
    <h1>
      Chart
    </h1>
  </Component>
</Component>
`;
